### PR TITLE
Removes redeeming mining points without a silo

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Alt to https://github.com/tgstation/tgstation/pull/67624

If you put materials in an Ore Redemption Machine while it isn't synced to an Ore Silo, the ORM will hold the mining points, and you won't be able to claim it until you sync the machine up.
I wanted to do this from the start (though originally I just wanted to halve the points), but didn't want to affect ghost roles, ~~but I was told that isn't a problem.~~ I was lied to, but I gave Golems an Ore silo in this PR.

https://user-images.githubusercontent.com/53777086/172668714-c04cd516-2a9f-4d4a-841f-18a84220f096.mp4

## Why It's Good For The Game

1- I found this was the better choice rather than halving points/any other solution because Miners will have to bring the ORM up and down with them (and sync every time they bring it up) whenever they want to claim points, which at that point isn't worth the hassle compared to just leaving it up.

2- Being able to bring the ORM down to no negative consequence to you, and only to the station, means that malicious players can screw over the station and their needs for solely themselves.
The Quartermaster CAN demote them, if they are able to catch the Miner who can run across lavaland, but I don't see that as a permanent solution.

3- With it being a Traitor objective, Traitors will often deconstruct it, then rebuild it to cover their tracks. This causes unknowing Shaft Miners to drop their mats in the ORM, unaware that the station is not being provided with it. Removing their ability to get points out of it will not only immediately let them know about the problem, but encourage them to get it fixed ASAP. This makes ORM sabotage more noticable, and more likely to get repaired, before the station starts complaining to Miners who don't know any better.

## Changelog

:cl:
qol: the Golem ship now spawns with an ore silo board and the necessary equipment to build it.
balance: ORMs will not allow you to claim points if it's not synced to an Ore Silo.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
Why is this good for the game? You only explain why it's better than other options, why does it need to exist at all?

File: code/modules/mining/machine_redemption.dm [hunk 1]
```diff
@@ -250,6 +251,9 @@
 	var/datum/component/material_container/mat_container = materials.mat_container
 	switch(action)
 		if("Claim")
+			if(!materials.silo)
+				to_chat(usr, span_warning("Mining points undistributable while unsynced to an ore silo."))
```
Comment (Fikou) [inline code/modules/mining/machine_redemption.dm:255]: You need mining points to set up RnD as a golem.
You need RnD as a golem to create an ore silo.

**Comment by Redrover1760** [comment]
Why are the upvotes and downvotes removed? Also, please also consider this in the context of the other pr that removes orms from being connectable from lavaland to the station ore silo.

Because there is 0 malicious consequences for the station for bringing the ORM to mining. You can literally walk through and print materials using any protolathe. You are literally insane. For claiming that miners bringing the ORM to mining harms the game in any way. I have never been bugged or demoted or anything for doing this, and I have been doing this for a long time.

**Comment by theta060** [comment]
half the time on mid-high pop miners dont mine die or are just new this just takes the issue further

**Comment by JohnFulpWillard** [comment]
> half the time on mid-high pop miners dont mine die or are just new this just takes the issue further

How does it take it further? If anything I find this to be better for those scenarios since someone can unsync the ORM and then players are stuck putting ores in an ORM not connected to the station, not being able to figure out why the station is complaining about the lack of mats.

**Comment by Shadowflame909** [comment]
Can we stop to ask why another miner rebalance that tries to solve the same issue that the one which got merged yesterday already solved?

https://github.com/tgstation/tgstation/pull/66953

No kneejerk alternative to goofs PR was needed. Just close it, since it feels like a chain reaction before this last one has settled.

**Comment by theta060** [comment]
> > half the time on mid-high pop miners dont mine die or are just new this just takes the issue further
> 
> How does it take it further? If anything I find this to be better for those scenarios since someone can unsync the ORM and then players are stuck putting ores in an ORM not connected to the station, not being able to figure out why the station is complaining about the lack of mats.

now miners have to go all the way on station needlessly to put in ores because you removed lavaland orm this is pretty much part of the PR that fucked it in the first place

**Comment by the-orange-cow** [comment]
A better solution to this PR would be reverting https://github.com/tgstation/tgstation/pull/66953, which created unnecessary inconvenience.

**Comment by JohnFulpWillard** [comment]
> now miners have to go all the way on station needlessly to put in ores because you removed lavaland orm this is pretty much part of the PR that fucked it in the first place

Lavaland never had an ORM

> No kneejerk alternative to goofs PR was needed. Just close it, since it feels like a chain reaction before this last one has settled.

Like I said in the PR body, I didn't do this in my original PR because I didn't want it to affect ghost roles. I spoke with goof about it the morning, before he actually made the PR. I made mine later in the day because
1- I wanted to do it regardless of what goof did
2- Never asked goof to open theirs
3- I wasn't at my PC to make my PR before goof did.

**Comment by Iamgoofball** [comment]
:+1:

**Comment by theta060** [comment]
> > now miners have to go all the way on station needlessly to put in ores because you removed lavaland orm this is pretty much part of the PR that fucked it in the first place
> 
> Lavaland never had an ORM
> 
you really have never seen a miner at all? anyone with a brain took the orm to lavaland because its useless on station

**Comment by Elli-Skala** [comment]
Wouldn't this mean golems (who don't start with a ore silo) could never get points, and in turn never get the protolathe boards, which you would use to even get a silo

**Comment by JohnFulpWillard** [comment]
> Wouldn't this mean golems (who don't start with a ore silo) could never get points, and in turn never get the protolathe boards, which you would use to even get a silo

Fair in assuming so since I forgot to add it to the body, but I did.
image

**Comment by SkeletalElite** [comment]
Reducing QoL of a job to fix a problem you created, nice

Should revert the Z level restriction instead. If you don't want the station ORM moved from the station to lavaland where it's more conveniant for the main job who use it should've mapped in a second one instead.

**Comment by GuillaumePrata** [comment]
> Reducing QoL of a job

Calling never interacting with the main station while you sit down there at your corner of the world as QoL made me laugh at least.

**Comment by SkeletalElite** [comment]
> > Reducing QoL of a job
> 
> Calling never interacting with the main station while you sit down there at your corner of the world as QoL made me laugh at least.

miners moving the ORM to right outside the mining shuttle, which is what they will do after this change, is not going to increase their interaction with anyone on the station. Its just going to waste time flying up and back down every time they need to deposit

**Comment by JohnFulpWillard** [comment]
> miners moving the ORM to right outside the mining shuttle, which is what they will do after this change, 

Why are they only going to start doing it now, after this change? What do they do currently? Because they already can't bring the ORM down without disconnecting it.

> It may surprise you to find out people who sign up for the miner job do so to interact with the mining job content, 

Shaft Miners are a job within supply, so being on the station's cargo department to drop mats off is still their 'content'.

> Requiring them to visit the station for 10 seconds + 30 seconds of shuttle fly a few times a shift doesn't change that.

Then what's the problem?

**Comment by the-orange-cow** [comment]
> Why are they only going to start doing it now, after this change? What do they do currently? Because they already can't bring the ORM down without disconnecting it.

Because they can't bring it down to lavaland anymore.  Right outside the shuttle will be the best pot for it until your next pr that makes it unmovable or something.



> Then what's the problem?

You are introducing arbitrary and pointless time wasting for questionable reasons, and people find this unsatisfying.

**Comment by SkeletalElite** [comment]
> > Requiring them to visit the station for 10 seconds + 30 seconds of shuttle fly a few times a shift doesn't change that.
> 
> Then what's the problem?

making miners go back to the station is literally just a waste of their time that adds no value to the game.

**Comment by Shadowflame909** [comment]
The arguments here is kind of why mining design is shifting towards keeping it on the station z-level. With, a sort of mind-link to the mechs on the lavaland z-level.

It's so that antags and other agents can still reach you without you necessarily being an unkillable monster or killed by a monster in a dangerous area they dont have the tools to reach/beat

This is definitely a good inbetween for the rounds I have to kidnap/kill/brainwash the shaftminer. I know where I can safely get the drop on them.

**Comment by SkeletalElite** [comment]
> This is definitely a good inbetween for the rounds I have to kidnap/kill/brainwash the shaftminer. I know where I can safely get the drop on them.

You can literally do that on that mining station, probably easier since there's more places to hide and more plausible deniability for random people being there and less witnesses. 

Miners are going to be moving the ORM to right outside the shuttle dock after this change. If a miner sees you hanging out all the way in the back room of cargo they are going to instantly know you are there specifically to kill a miner because there's pretty much no other reason to just be hanging around there.

**Comment by Palpetinus** [comment]
So the problem is:
Miners don't interact with the station enough

The proposed solution is:
Leave the ORM on the station z-level to force miners to go back to it

What we got is:
Miners not knowing the change, therefore never putting ores into the silo leaving them to float in the disconnected ORM
-----you are here-----
The now proposed solution to THAT is:
Don't give the miners any rewards for using a disconnected ORM

The future (and already sometimes current) problem is:
Miners just drag the ORM to the shuttle port. Miners STILL don't interact with the station.

Side note: Miners already visit cargo twice at a minimum, sometimes thrice: once to get the ORM roundstart, once to get the satchel of holding, once to get the plasmacutter. Sometimes both in one trip.

I have to ask.. what's the end goal? These changes feel like a kneejerk reaction with the assumption of "maybe something will be better" without the slightest bit of thinking about the end result involved.

**Comment by zxaber** [comment]
Maybe instead we should remove the mining vendors and make miners use their points to buy their things through cargo?

**Comment by Mooshimi** [comment]
> Why are the upvotes and downvotes removed? Also, please also consider this in the context of the other pr that removes orms from being connectable from lavaland to the station ore silo.

this was @JohnFulpWillard 's change anyways, the zlevel orm change

**Comment by Mooshimi** [comment]
> > > Requiring them to visit the station for 10 seconds + 30 seconds of shuttle fly a few times a shift doesn't change that.
> > 
> > 
> > Then what's the problem?
> 
> making miners go back to the station is literally just a waste of their time that adds no value to the game.

Them being an antag's objective

**Comment by Mooshimi** [comment]
Hear me out.... make satchels take longer to research

**Comment by Shadowflame909** [comment]
Nah that just hurts miners ability to contribute to the station with a lot of ores.

I think medical and other departments should get some features miners could use to make them stronger against simple mobs only. Departmental cooperation for a PVE buff

**Comment by Mooshimi** [comment]
> Nah that just hurts miners ability to contribute to the station with a lot of ores.
> 
> I think medical and other departments should get some features miners could use to make them stronger against simple mobs only. Departmental cooperation for a PVE buff

they already do with the healing viruses and augments/organ upgrades etc. it's just not worth it to come up and get them because round not long and healing pens

**Comment by SkeletalElite** [comment]
> Them being an antag's objective

as I wrote in my previous comment

"You can literally do that on that mining station, probably easier since there's more places to hide and more plausible deniability for random people being there and less witnesses."

**Comment by Iamgoofball** [comment]
>You can literally do that on that mining station

Yes, where the miner can walk 2 feet and have his roguelite boss fighting equipment at full power. Nice try.

**Comment by the-orange-cow** [comment]
> > You can literally do that on that mining station
> 
> Yes, where the miner can walk 2 feet and have his roguelite boss fighting equipment at full power. Nice try.

PKAs are worse than a stunbaton in most situations, and antags generally have access to superior gamer gear anyways.

**Comment by anclandor** [comment]
> Them being an antag's objective

Killing miners at mining base is easier than killing them on the station because you can easily disable tcomms for lavaland and get the jump on miners by hiding.

**Comment by JohnFulpWillard** [comment]
> Killing miners at mining base is easier than killing them on the station because you can easily disable tcomms for lavaland and get the jump on miners by hiding.

This PR isn't removing the ability to do that.

> Miners just drag the ORM to the shuttle port.

I don't see a problem with that?

> Maybe instead we should remove the mining vendors and make miners use their points to buy their things through cargo?

Imagine a society where you spend points on a chef produce console-type console but with mining equipment instead, but I also am too lazy to make that myself.

> Yes, where the miner can walk 2 feet and have his roguelite boss fighting equipment at full power. Nice try.

If they get back to lavaland, through the doors, despite your stealth and surprise attack, then sounds like skill issue. Though (again), this PR isn't removing the ability to do that, I don't see the problem.

**Comment by SkeletalElite** [comment]
> Miners are going out of their way to move the ORM to a place no one has access to, but still are getting no benefit to doing so because most of the reason they move it is so they don't have to use the cargo shuttle. What we're doing now is just **discouraging** them from doing it, that's my goal, and it seems to work because I don't see the ORM moved at all anymore from the rounds I've observed. The minority of rounds where that happens doesn't really bother me.

Why is the solution to discouraging miners from moving the ORM removing the point of moving it in the first place (saving time, QOL) instead of just mapping a second ORM

**Comment by datorangebottle** [comment]
Was effort put in to make sure this doesn't screw free golems over? Last I checked, they don't get a silo until they get the Legacy, which requires 2k points.

**Comment by moocowswag** [comment]
Making miners care about the station more means they bring their powergame lavaland loot to the station more and valid hunt anything that steps anywhere near cargo

I thought there was a growing sentiment that we wanted miners and lavaland in general to be more seperate from the station? Or did the fruit change his mind?.

**Comment by Mothblocks** [comment]
No maintainer has expressed interest in this PR

**Pull request closed without merging**